lib/publiccloud/provider.pm

```diff
@@ -537,6 +539,8 @@ sub terraform_apply {
         $cmd .= "-var 'region=" . $self->provider_client->region . "' ";
         $cmd .= "-var 'name=" . $self->resource_name . "' ";
         $cmd .= "-var 'project=" . $args{project} . "' " if $args{project};
+        my $cloud_init = get_var('PUBLIC_CLOUD_CLOUD_INIT');
+        $cmd .= "-var 'cloud_init=" . TERRAFORM_DIR . "/cloud-init.yaml' " if ($cloud_init);
```

Comment:
since we need it in several places , maybe we can just create provider's property ? something like : 
`has cloud_init => get_var('PUBLIC_CLOUD_CLOUD_INIT')`

and then just do : 
```suggestion
        $cmd .= "-var 'cloud_init=" . TERRAFORM_DIR . "/cloud-init.yaml' " if ($self->cloud_init); 
```
:thinking: 

P.S. this is evolution of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19969/files#r1720797254

Reply:
resolved